packages/backend-plugin-api/api-report.md

```diff
@@ -312,6 +324,29 @@ export interface DiscoveryService {
   getExternalBaseUrl(pluginId: string): Promise<string>;
 }
 
+// @public (undocumented)
+export interface EventParams<TPayload = unknown> {
+  eventPayload: TPayload;
+  metadata?: Record<string, string | string[] | undefined>;
```

Comment:
Same thing here. I do find it a bit awkward to have metadata attached to events. Should this not be in the event payload instead? The topic should define a strict schema for the event payload, but it wouldn't be the same for metadata right? perhaps they act a bit like annotations?

It's not how metadata is used right now though. It's more of an extra event payload that is adjacent to the "main" payload received from an external system. It could be that we want to properly frame it as such, but tbh I think the best approach here might be to remove the metadata and either move the fields into the payload, or handle them upfront when possible, for example in the case of sub topic routers. Perhaps it's something that we can keep in some interface for extension points for HTTP ingestion in the events backend, but not for the events service itself?

I know it's quite a big change, so curious on your initial thoughts @pjungermann